src/dotnet/GeneratorAgent/src/Azure.Tools.GeneratorAgent/Security/InputValidator.cs

```diff
@@ -26,225 +26,193 @@ internal static class InputValidator
         /// <param name="path">The path to validate</param>
         /// <param name="pathType">Description of the path type for logging</param>
         /// <returns>Result indicating if path is safe</returns>
-        public static Result<string> ValidateDirTraversal(string? path, string pathType = "path")
+        public static string ValidateDirTraversal(string? path, string pathType = "path")
         {
             if (string.IsNullOrWhiteSpace(path))
             {
-                return Result<string>.Failure(new ArgumentException($"{pathType} cannot be null or empty"));
+                throw new ArgumentException($"{pathType} cannot be null or empty");
             }
 
-            return Result<string>.Success(path);
+            return path;
         }
 
         /// <summary>
         /// Validates a TypeSpec path for security and format.
         /// </summary>
         /// <param name="path">The TypeSpec path to validate</param>
         /// <param name="isLocalPath">True if this is a local filesystem path, false if it's a relative repository path</param>
-        public static Result<string> ValidateTypeSpecDir(string? path, bool isLocalPath = true)
+        public static string ValidateTypeSpecDir(string? path, bool isLocalPath = true)
```

Comment:
nit but since the method actually makes some changes to the path maybe it's worth naming it ValidateAndNormalizeTypeSpecDir or something like that

There are also quite a few of the same thing in this file, this is definitely a nit but just in my opinion only including validate in the name makes it feel like it's just doing checks and not changing anything